src/System/Configuration.cpp

```diff
@@ -165,9 +165,11 @@ namespace ame
         FETCH(MapNames, rom.readPointer());
         FETCH(MapNameCount, rom.readByte());
         FETCH(MapNameTotal, rom.readByte());
-        OverworldCount++; // value is one less
-        ItemCount *= 2;
-        MapNameCount += (RomType == RT_FRLG ? 1 : 0); // correct FRLG
+        // the values from the ROM are the max indices, so we have to add 1
+        PokemonCount++;
+        OverworldCount++;
+        ItemCount = ItemCount * 2 + 1;
+        MapNameCount++;
```

Comment:
I knew that it's the same for Emerald, but your right in that RS differs. I updated the commit.